lib/mp4/transmuxer.js

```diff
@@ -887,17 +887,17 @@ CoalesceStream.prototype.flush = function(flushSource) {
 
   initSegment = mp4.initSegment(this.pendingTracks);
 
-  // Create a new typed array large enough to hold the init
-  // segment and all tracks
-  event.data = new Uint8Array(initSegment.byteLength +
-                              this.pendingBytes);
+  // Create a new typed array to hold the init segment
+  event.initSegment = new Uint8Array(initSegment.byteLength);
```

Comment:
Is it worth it to try to pass down whether we want the initSegment created or not, or is having to maintain that state too costly/is the creation and passing of the init segment cheap enough that it isn't a big deal (or is there another concern I'm missing)?